engine/src/main/java/org/pentaho/di/repository/kdr/delegates/metastore/KettleDatabaseRepositoryMetaStore.java

```diff
@@ -0,0 +1,460 @@
+      for ( RowMetaAndData elementRow : elementRows ) {
+        IMetaStoreElement element = delegate.parseElement( elementType, elementRow );
+        elements.add( element );
+      }
+      return elements;
+    } catch ( Exception e ) {
+      throw new MetaStoreException( "Unable to get list of elements from namespace '"
+        + namespace + "' and for element type '" + elementType.getName() + "'", e );
+    }
+  }
+
+  @Override
+  public List<String> getElementIds( String namespace, IMetaStoreElementType elementType ) throws MetaStoreException {
+    List<String> ids = new ArrayList<String>();
+    List<IMetaStoreElement> elements = getElements( namespace, elementType );
+    for ( IMetaStoreElement element : elements ) {
+      ids.add( element.getId() );
+    }
+    return ids;
+  }
+
+  @Override
+  public IMetaStoreElement getElement( String namespace, IMetaStoreElementType elementType, String elementId ) throws MetaStoreException {
+    try {
+      RowMetaAndData elementRow = delegate.getElement( new LongObjectId( new StringObjectId( elementId ) ) );
+      if ( elementRow == null ) {
+        return null;
+      }
+      return delegate.parseElement( elementType, elementRow );
+    } catch ( Exception e ) {
+      throw new MetaStoreException( "Unable to get element", e );
+    }
+  }
+
+  @Override
+  public IMetaStoreElement getElementByName( String namespace, IMetaStoreElementType elementType, String name ) throws MetaStoreException {
+    try {
+      LongObjectId namespaceId = delegate.getNamespaceId( namespace );
+      if ( namespaceId == null ) {
+        return null;
+      }
+      LongObjectId elementTypeId = delegate.getElementTypeId( namespaceId, elementType.getName() );
+      if ( elementTypeId == null ) {
+        return null;
+      }
+      LongObjectId elementId = delegate.getElementId( elementTypeId, name );
+      if ( elementId == null ) {
+        return null;
+      }
+      RowMetaAndData elementRow = delegate.getElement( elementId );
+      if ( elementRow == null ) {
+        return null;
+      }
+
+      return delegate.parseElement( elementType, elementRow );
+    } catch ( Exception e ) {
+      throw new MetaStoreException( "Unable to get element by name '"
+        + name + "' from namespace '" + namespace + "'", e );
+    }
+  }
+
+  @Override
+  public void createElement( String namespace, IMetaStoreElementType elementType, IMetaStoreElement element ) throws MetaStoreException, MetaStoreElementExistException {
+    try {
+      IMetaStoreElement found = getElementByName( namespace, elementType, element.getName() );
+      if ( found != null ) {
+        throw new MetaStoreElementExistException( Arrays.asList( found ), "The element with name '"
+          + element.getName() + "' already exists" );
+      }
+
+      delegate.insertElement( elementType, element );
+      repository.commit();
+    } catch ( MetaStoreElementExistException e ) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An exception is caught and re-thrown within the catch clause without any other operations being performed on the exception, making the catch clause unnecessary.

**Fix**: Perform additional operations on the exception that has been caught. You can wrap a standard exception into a custom exception to add additional information or special handling. When you do that, make sure to set the original exception as the cause. Otherwise, you will lose the stack trace and message of the original exception, which will make it difficult to analyze the event that caused the exception.  
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)